x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/logic/diff/calculation/algorithms/force_target_version_diff_algorithm.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  ThreeVersionsOf,
+  ThreeWayDiff,
+} from '../../../../../../../../common/api/detection_engine/prebuilt_rules';
+import {
+  MissingVersion,
+  ThreeWayDiffConflict,
+  ThreeWayDiffOutcome,
+  ThreeWayMergeOutcome,
+} from '../../../../../../../../common/api/detection_engine/prebuilt_rules';
+
+/**
+ * Diff algorithm forcing target version. Useful for special fields like `version`.
+ */
+export const forceTargetVersionDiffAlgorithm = <TValue>(
+  versions: ThreeVersionsOf<TValue>
+): ThreeWayDiff<TValue> => {
+  const {
+    base_version: baseVersion,
+    current_version: currentVersion,
+    target_version: targetVersion,
+  } = versions;
+  const hasBaseVersion = baseVersion !== MissingVersion;
+
+  return {
+    has_base_version: hasBaseVersion,
+    base_version: hasBaseVersion ? baseVersion : undefined,
+    current_version: currentVersion,
+    target_version: targetVersion,
+    merged_version: targetVersion,
+    merge_outcome: ThreeWayMergeOutcome.Target,
+
+    diff_outcome: hasBaseVersion
+      ? ThreeWayDiffOutcome.StockValueCanUpdate
+      : ThreeWayDiffOutcome.MissingBaseCanUpdate,
```

Comment:
We also need to handle a situation when there's no update for proper diffing without a version bump. I.e, the case when a rule is edited locally rather than compared to a remote version.

Reply:
I applied `determineDiffOutcome()` it looks like a good fit here.